.github/workflows/update-snarkvm.yml

```diff
@@ -0,0 +1,163 @@
+name: Update snarkVM dependency
+
+on:
+  schedule:
+    # Run every hour to check for new commits in snarkVM staging
+    - cron: '0 * * * *'
+  push:
+    branches:
+      - 'ci/snarkvm-update'
+
+env:
+  RUST_BACKTRACE: 1
+
+jobs:
+  update-snarkvm-staging:
+    name: Update snarkVM to latest staging
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout snarkOS
+        uses: actions/checkout@v5
+        with:
+          ref: staging
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get latest snarkVM staging commit
+        id: snarkvm-commit
+        uses: actions/github-script@v7
+        with:
+          script: |
+            // Get the latest commit hash from snarkVM staging branch
+            const { data: branch } = await github.rest.repos.getBranch({
+              owner: 'ProvableHQ',
+              repo: 'snarkVM',
+              branch: 'staging'
+            });
+            
+            const latestCommit = branch.commit.sha;
+            console.log('Latest snarkVM staging commit:', latestCommit);
+            core.setOutput('latest_commit', latestCommit);
```

Comment:
Not entirely sure what current state on these is as the post is old but `setOutput` was slated to be deprecated: https://github.blog/changelog/2023-07-24-github-actions-update-on-save-state-and-set-output-commands/